### PR TITLE
fix result dictionary

### LINKED ISSUE #135: fix results dictionary

### PR DESCRIPTION
**Pull request by federicazanca**
close #135

### CODE REVIEW

File: aiida_mlip/parsers/sp_parser.py [hunk 1]
```diff
@@ -98,7 +98,9 @@ def parse(self, **kwargs) -> int:
         with self.retrieved.open(xyz_output, "rb") as handle:
             self.out("xyz_output", SinglefileData(file=handle, filename=xyz_output))
 
-        content = read(Path(self.node.get_remote_workdir(), xyz_output))
+        content = read(
+            Path(self.node.get_remote_workdir(), xyz_output), format="extxyz"
```
Comment (ElliottKasoar) [inline aiida_mlip/parsers/sp_parser.py:102]: Why is this necessary?

In general janus allows any file format to be written out. Doesn't this limit that?

**Comment by ElliottKasoar** [comment]
The test errors may well be related to ASE 3.23.0

**Review by ElliottKasoar — COMMENTED** [review]
It might be nice to add a test that would fail before this change.

File: aiida_mlip/parsers/sp_parser.py (hunk 1 above)
Comment (federicazanca) [reply aiida_mlip/parsers/sp_parser.py:102]: yeah I thought it would be only xyz files. 
but even if I didn't do this, I'm still reading it as a dictionary, if the output was another format I couldn't do that (? depends on the format I guess)

**Comment by federicazanca** [comment]
> It might be nice to add a test that would fail before this change.



> It might be nice to add a test that would fail before this change.



> It might be nice to add a test that would fail before this change.



> It might be nice to add a test that would fail before this change.



> It might be nice to add a test that would fail before this change.

Yeah I'm trying to do this, I think actually maybe the failing test is because of this change (maybe)

File: aiida_mlip/parsers/sp_parser.py (hunk 1 above)
Comment (ElliottKasoar) [reply aiida_mlip/parsers/sp_parser.py:102]: I'm not actually sure how many alternative file formats allow you to write out data from `info`/`arrays`/`calc.results`, but in general if it's stored, it should be readable again.

You can certainly get an `info` dictionary after reading a .cif file, although I'm not sure about how much can be written in the first place.

If it does only make sense to support extxyz, this should documented and probably (`--out`) validated.

**Comment by federicazanca** [comment]
> The test errors may well be related to ASE 3.23.0

On second thought this is probably the cause of all the problems

File: tests/calculations/configs/config_janus.yaml [hunk 2]
```diff
@@ -1,4 +1,3 @@
 properties:
   - "energy"
-out: "results.xyz"
```
Comment (ElliottKasoar) [inline tests/calculations/configs/config_janus.yaml:3]: Does this not work, or do you just not want to test it anymore?

Comment (federicazanca) [reply tests/calculations/configs/config_janus.yaml:3]: I removed it just to test something. But I think it should stay removed, cause the way the code is written at the moment it's always gonna use the default name for the output (which is what we want cause it's easier to handle for us and the user should not interact with the file directly anyways, so we know the output is always a xyz)

Comment (ElliottKasoar) [reply tests/calculations/configs/config_janus.yaml:3]: So this isn't actually used at the moment anyway?

If you want to ensure the output is an xyz, you can also specify the format, which would also override any file extensions, but maybe that could be more confusing.
